### PR TITLE
fix 'In List' and 'Equal to' filters for number field

### PR DESCRIPTION
**Pull request by sunm19810-pki**
The `In list` and `Equal to` will not work for number filed:
image

### CODE REVIEW

File: src/DataTable/utils/queryParams.js [hunk 1]
```diff
@@ -194,7 +194,11 @@ function getEntitiesForGivenFilter(entities, filter, ccFields, ownProps) {
   const { filterOn, filterValue, selectedFilter } = filter;
   const field = ccFields[filterOn];
   const { path, getValueToFilterOn } = field;
-  const subFilter = getSubFilter(false, selectedFilter, filterValue);
+  const subFilter = getSubFilter(
+    false,
+    selectedFilter,
+    field.type === "number" ? Number(filterValue) : filterValue
```
Comment (sunm19810-pki) [inline src/DataTable/utils/queryParams.js:200]: fix `Equal To` for number field

File: src/DataTable/utils/queryParams.js [hunk 2]
```diff
@@ -302,11 +306,11 @@ function getSubFilter(
       ? qb.inList(arrayFilterValue) //filter using qb (aka we're backend connected)
       : fieldVal => {
           //filter using plain old javascript (aka we've got a local table that isn't backend connected)
-          if (!fieldVal || !fieldVal.toLowerCase) return false;
+          if (!fieldValue?.toString) return false;
           return (
             arrayFilterValue
               .map(val => val && val.toLowerCase())
-              .indexOf(fieldVal.toLowerCase()) > -1
+              .indexOf(fieldVal.toString().toLowerCase()) > -1
```
Comment (sunm19810-pki) [inline src/DataTable/utils/queryParams.js:313]: fix `In list` for number field

**Comment by tnrich** [comment]
Hey @sunm19810-pki thanks for this PR! Looks like it should do what you want, but do you think you could set up a simple cypress test confirming that this functionality works?

Thanks!

**Review by tnrich — APPROVED** [review]

**Comment by tnrich** [comment]
Thanks @sunm19810-pki !

**Pull request merged**